### DISCUSSION TITLE
Feature Request: Custom Model Name Support for On-Premise Anthropic AP

### DISCUSSION THREAD

**Discussion by santosha001** [cline/cline#941] [Feature Requests]
Add support for custom model names when using on-premise Anthropic API with the Cline VS Code extension.
Description: Currently, the Cline extension provides a dropdown for model selection. However, when using an on-premise Anthropic API, our model names differ slightly from those listed. To address this, we propose adding a feature that allows users to overwrite the model name, enabling the use of custom models as required.
Proposed Solution:
	1.	Add a new configuration option in the extension settings to enable custom model name input.
	2.	When enabled, provide a text input field next to or below the model selection dropdown.
	3.	Allow users to enter their custom model name in this field.
	4.	Use the custom model name for API requests when specified, otherwise default to the selected dropdown option.
Use Case: This feature would benefit organizations using on-premise Anthropic API setups with customized model names, ensuring compatibility and seamless integration with their existing infrastructure.

**Comment by corporate9601** [comment]
+1 I need this today since Claude 3.7 just dropped and the Cline VS Code extension has the aforementioned dropdown list for model selection. Needs custom input!
